js/index.js

```diff
@@ -17,3 +17,125 @@ for (let i = 0; i < skills.length; i++) {
     skill.innerText = skills[i];
     skillsList.appendChild(skill);
 }
+
+
+const messageForm = document.querySelector('form[name="leave_message"]');
+const messageSection = document.querySelector('#messages');
+const messageList = messageSection.querySelector('ul');
+
+messageSection.hide = function () {
+    this.style.display = 'none';
+};
+messageSection.show = function () {
+    this.style.display = '';
+};
+
+Object.defineProperty(messageList, "hasNoMessages", {
+    get: function () {
+        return this.childElementCount === 0
+    }
+});
+
+function updateMessageSection() {
+    if (messageList.hasNoMessages) {
+        messageSection.hide();
+    }
+};
+updateMessageSection();
+
+
+messageForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const form = e.target;
+
+    const userName = form.usersName.value;
+    const useEmail = form.usersEmail.value;
+    const userMessage = form.usersMessage.value;
+
+    // console.log(userName);
+    // console.log(useEmail);
+    // console.log(userMessage);
+
+    const newMessage = document.createElement('li');
+    newMessage.innerHTML = `
+        <a href="mailto:${useEmail}" target="_blank">${userName}</a>
+        wrote: <span> ${userMessage}</span>
+    `;
+    newMessage.messageText = userMessage;
+
+    const removeButton = document.createElement('button');
+    removeButton.innerText = "remove";
+    removeButton.type = "button";
+
+    const editButton = document.createElement('button');
+    editButton.innerText = "edit";
+    editButton.type = "button";
+
+
+    newMessage.appendChild(editButton);
+    newMessage.appendChild(removeButton);
+    messageList.appendChild(newMessage);
+    messageSection.show();
+
+    form.reset();
+});
+
+messageList.addEventListener('click', (e) => {
+    if (e.target.tagName === 'BUTTON') {
```

Comment:
because a large conditional like this causes the entire rest of the function to be `{}` nested and indented, it can sometimes make things a little bit harder to read. 

Using a "guard statement" in a case like this can be a nice way to reduce the visual complexity of the nesting: 

rather than 

```
if (condition) {
  // doing the entire rest of the function inside the conditional scope/block
}
```
we can invert the condition and only put a single line of code inside of the brackets: 

```
if (!condition) {
  // break out of the function early
  return;
}

// because the `if` contains a `return` statement, we can put what would be the `else` right after it with no nesting, so the entire rest of the function becomes de-nested by one level:

// do all the rest of the function stuff at the "top" or "outermost" scope/block of the function
```
This is very much a matter of style, and some people prefer one way over the other, but I just wanted to point it out =D